### ISSUE TITLE
Trouble with animating newly added items

### ISSUE THREAD

**Issue by aorozsa** [haltu/muuri#329]
Overall, I'm trying to achieve what's done in the demo: newly created items popping in with an animation instead of instantly appearing. This is the function I've created for adding in one card at a time.

`function addItem(element) {`
`  grid.add(element, {`
`    layout: function(i) {`
`      var items = grid.getItems();`
`      items = items[items.length - 1];`
`      console.log(content(items));`
`      grid.hide(items, {`
`        instant: true,`
`        onFinish: function(hiddenItems) {`
`          grid.show(hiddenItems);`
`        }`
`      });`
`    }`
`  });`
`}`

It works for the most part; adding in new items works like a charm. The problem is when loading in saved configurations. When there are three or more items, the second item always fails to animate. It doesn't matter how many cards there are, it's always the second one. In a nutshell, initialisation works by creating the grid, creating the static first card's markup and adding it using the above function, loading autosave data from window.localstorage, parsing it, and looping through every item in it, creating item templates based on the items and passing that to the above function. Below is a recording that demonstrates refreshing the page with no autosaved data, adding in a card, refreshing, adding in more cards, refreshing again, changing the values on the cards so that they're distinct, and refreshing one last time.

[problemDemo.zip](https://github.com/haltu/muuri/files/3563339/problemDemo.zip)

The console output shows that "1" is skipped entirely, and "11" is repeated twice.

**Comment by aorozsa** [comment]
I did some more testing, and the layout function is being skipped entierly on that first card. So, addItem is called, but the layout function inside isn't. I even tried temporarily removing the declaration of the static first card since it's created in a slightly different way, but that didn't change anything.

I was able to change addItem to accept arrays of new elements to add instead of just one (which also required me to modify the two other necessary functions used in loading). So, instead of adding one card at a time, the load function in my code places the elements in an array and then adds them all at once. This fixed my issue, but I feel that it's an unnecessary complication to the code (unless it's impossible to find out why the second layout call is skipped). If there are any ideas for how this bizarre bug can be fixed, I'd love to hear them. Thanks.
